app/forms/idv/doc_pii_form.rb

```diff
@@ -0,0 +1,51 @@
+module Idv
+  class DocPiiForm
+    include ActiveModel::Model
+
+    validates_presence_of :first_name
+    validates_presence_of :last_name
+    validates_presence_of :dob
+
+    attr_reader :first_name, :last_name, :dob
+
+    def initialize(pii)
+      @first_name = pii[:first_name]
+      @last_name = pii[:last_name]
+      @dob = pii[:dob]
+    end
+
+    def submit
+      FormResponse.new(
+        success: valid?,
+        errors: error_hash,
+        extra: { original_errors: errors },
```

Comment:
Would it have been possible to avoid `error_hash` and just use ActiveModel validations instead?

In other forms handled by `ImageUploadResponsePresenter`, it's expecting `errors` as an instance of `ActiveModel::Errors`. For current usage, it should be fine, but could maybe cause some confusion if we're expecting methods of the errors object to be available (not in the plain hash here).

Is `original_errors` used anywhere?

Reply:
The issue was it adds errors to `:last_name` and `:dob` and ending up with:

![image](https://user-images.githubusercontent.com/1430443/97882501-4c6b7300-1ce9-11eb-9f1c-c6e13a5aea81.png)

The weekend has given me the idea to run custom validator functions, and use `add_error` instead?

I don't know if there's a good way to unify the message we show with the programmatic support of being able to know it was first_name AND last_name, just first_name, etc. I added `original_errors` as I thought the form response got used by analytics somewhere.